rioxarray/_io.py

```diff
@@ -1004,36 +1029,43 @@ def open_rasterio(
     data = indexing.CopyOnWriteArray(data)
     if cache and chunks is None:
         data = indexing.MemoryCachedArray(data)
-
-    result = DataArray(
-        data=data, dims=(coord_name, "y", "x"), coords=coords, attrs=attrs, name=da_name
-    )
+    if output_is_dataset:
+        # Remove band specific attrs from Dataset attrs
+        band_attrs = riods.tags(1)
+        for k in band_attrs.keys():
+            attrs.pop(k)
+        result = coords.assign_attrs(attrs)
+    else:
+        result = DataArray(
+            data=data, dims=(coord_name, "y", "x"), coords=coords, attrs=attrs, name=da_name
+        )
     result.encoding = encoding
-
     # update attributes from NetCDF attributess
     _load_netcdf_attrs(riods.tags(), result)
     result = _decode_datetime_cf(
         result, decode_times=decode_times, decode_timedelta=decode_timedelta
     )
-
     # make sure the _FillValue is correct dtype
-    if "_FillValue" in result.attrs:
+    if "_FillValue" in result.attrs and not output_is_dataset:
         result.attrs["_FillValue"] = result.dtype.type(result.attrs["_FillValue"])
-
     # handle encoding
     _handle_encoding(result, mask_and_scale, masked, da_name, unsigned=unsigned)
     # Affine transformation matrix (always available)
     # This describes coefficients mapping pixel coordinates to CRS
     # For serialization store as tuple of 6 floats, the last row being
     # always (0, 0, 1) per definition (see
     # https://github.com/sgillies/affine)
-    result.rio.write_transform(riods.transform, inplace=True)
-    rio_crs = riods.crs or result.rio.crs
-    if rio_crs:
-        result.rio.write_crs(rio_crs, inplace=True)
+    # If we have a Dataset with variables and each variable has already 'grid_mapping: spatial_ref'
```

Comment:
I recommend not adding the CRS to each variable as you did previously and instead add it to the dataset.